src/AutoRest.CSharp/Common/Output/Models/ClientFields.cs

```diff
@@ -55,17 +55,17 @@ private ClientFields(IEnumerable<Parameter> parameters, BuildContext? context)
                 {
                     switch (scheme)
                     {
-                        case AzureKeySecurityScheme azureKeySecurityScheme:
-                            AuthorizationHeaderConstant = new(Private | Const, typeof(string), "AuthorizationHeader", $"{azureKeySecurityScheme.HeaderName:L}");
+                        case KeySecurityScheme keySecurityScheme:
+                            AuthorizationHeaderConstant = new(Private | Const, typeof(string), "AuthorizationHeader", $"{keySecurityScheme.Name:L}");
                             _keyAuthField = new(Private | ReadOnly, KnownParameters.KeyAuth.Type.WithNullable(false), "_" + KnownParameters.KeyAuth.Name);
 
                             fields.Add(AuthorizationHeaderConstant);
                             fields.Add(_keyAuthField);
                             credentialFields.Add(_keyAuthField);
                             parameterNamesToFields[KnownParameters.KeyAuth.Name] = _keyAuthField;
                             break;
-                        case AADTokenSecurityScheme aadTokenSecurityScheme:
-                            ScopesConstant = new(Private | Static | ReadOnly, typeof(string[]), "AuthorizationScopes", $"new string[]{{ {aadTokenSecurityScheme.Scopes.GetLiteralsFormattable()} }}");
+                        case OAuth2SecurityScheme oAuth2SecurityScheme:
+                            ScopesConstant = new(Private | Static | ReadOnly, typeof(string[]), "AuthorizationScopes", $"new string[]{{ {oAuth2SecurityScheme.Scopes.GetLiteralsFormattable()} }}");
```

Comment:
Can we have a test project with oauth2 or apiKey credential type and unit tests to verify the generated code?

Reply:
Added